dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/resourcetable/DefaultResourceTableService.java

```diff
@@ -255,4 +258,29 @@ public void dropAllSqlViews( JobProgress progress )
         progress.startingStage( "Drop SQL views", nonQueryViews.size(), SKIP_ITEM );
         progress.runStage( nonQueryViews, SqlView::getViewName, sqlViewService::dropViewTable );
     }
+
+    @Override
+    public List<Integer> generateDataYears()
```

Comment:
Would be nice to have a unit test for this public method covering the possible scenarios.